### DISCUSSION TITLE
[Enhancement] [tabs] Drag enable tabs

### DISCUSSION THREAD

**Discussion by cloudsolace** [element-plus/element-plus#15696] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `tabs`

## Description

It will be great if the tab component could be draggable.

### Additional comments

_(empty)_

**Comment by BulatSa** [comment]
It's good idea, because [vue.draggable.next](https://github.com/SortableJS/vue.draggable.next) and [vue-draggable-plus](https://github.com/Alfred-Skyblue/vue-draggable-plus) has bugs with order of tabs by adding
https://github.com/SortableJS/vue.draggable.next/issues/80
https://github.com/Alfred-Skyblue/vue-draggable-plus/issues/235

**Comment by prherranz** [comment]
Is there any updates from this topic? Will be nice to have the option to enable tabs header drag

**Comment by xingyixiang** [comment]
This feature can be implemented using `sortablejs`. use element-plus@2.9.10 I wrote an [demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtdGFicyB2LW1vZGVsPVwiYWN0aXZlTmFtZVwiIHJlZj1cInRhYlJlZlwiIGNsYXNzPVwiZGVtby10YWJzXCI+XG4gICAgPGVsLXRhYi1wYW5lIGxhYmVsPVwiVXNlclwiIG5hbWU9XCJmaXJzdFwiPlVzZXI8L2VsLXRhYi1wYW5lPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIkNvbmZpZ1wiIG5hbWU9XCJzZWNvbmRcIj5Db25maWc8L2VsLXRhYi1wYW5lPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIlJvbGVcIiBuYW1lPVwidGhpcmRcIj5Sb2xlPC9lbC10YWItcGFuZT5cbiAgICA8ZWwtdGFiLXBhbmUgbGFiZWw9XCJUYXNrXCIgbmFtZT1cImZvdXJ0aFwiPlRhc2s8L2VsLXRhYi1wYW5lPlxuICAgIDxlbC10YWItcGFuZSBsYWJlbD1cIkdyb3VwXCIgbmFtZT1cImZpZnRoXCIgZGlzYWJsZWQ+R3JvdXA8L2VsLXRhYi1wYW5lPlxuICA8L2VsLXRhYnM+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmLCBvbkJlZm9yZVVubW91bnQsIG9uTW91bnRlZCB9IGZyb20gJ3Z1ZSdcbmltcG9ydCBTb3J0YWJsZSBmcm9tICdzb3J0YWJsZWpzJ1xuaW1wb3J0IHsgdXNlR2V0RGVyaXZlZE5hbWVzcGFjZSB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB0eXBlIHsgVGFic0luc3RhbmNlIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5jb25zdCBhY3RpdmVOYW1lID0gcmVmKCdmaXJzdCcpXG5jb25zdCBucyA9IHVzZUdldERlcml2ZWROYW1lc3BhY2UoKS52YWx1ZVxubGV0IHNvcnRhYmxlOiBTb3J0YWJsZSB8IG51bGwgPSBudWxsXG5jb25zdCB0YWJSZWYgPSByZWY8VGFic0luc3RhbmNlPigpXG5vbk1vdW50ZWQoKCkgPT4ge1xuICBjb25zdCB7IHRhYkxpc3RSZWYsIHRhYkJhclJlZiB9ID0gdGFiUmVmLnZhbHVlIS50YWJOYXZSZWYhXG4gIHNvcnRhYmxlID0gbmV3IFNvcnRhYmxlKHRhYkxpc3RSZWYhLCB7XG4gICAgYW5pbWF0aW9uOiAxNTAsXG4gICAgZHJhZ2dhYmxlOiBgLiR7bnN9LXRhYnNfX2l0ZW1gLFxuICAgIGZpbHRlcjogJy5pcy1kaXNhYmxlZCcsXG4gICAgb25DaGFuZ2U6IChldmVudCkgPT4ge1xuICAgICAgLy8gSWYgRWx0YWJzIGhhcyBhIGJhciwgdGhlIHRhYiBiYXIgbmVlZHMgdG8gYmUgY2xlYXJlZCBmcm9tIFNvcnRhYmxlIGFuaW1hdGlvbnNcbiAgICAgIChzb3J0YWJsZSBhcyBhbnkpLnJlbW92ZUFuaW1hdGlvblN0YXRlKHRhYkJhclJlZj8uJGVsKVxuICAgICAgdGFiQmFyUmVmPy51cGRhdGUoKVxuXG4gICAgICBjb25zdCB7IG9sZERyYWdnYWJsZUluZGV4LCBuZXdEcmFnZ2FibGVJbmRleCB9ID0gZXZlbnRcbiAgICAgIGNvbnNvbGUubG9nKCdvbGREcmFnZ2FibGVJbmRleDonLCBvbGREcmFnZ2FibGVJbmRleClcbiAgICAgIGNvbnNvbGUubG9nKCduZXdEcmFnZ2FibGVJbmRleDonLCBuZXdEcmFnZ2FibGVJbmRleClcbiAgICB9XG4gIH0pXG59KVxuXG5vbkJlZm9yZVVubW91bnQoKCkgPT4ge1xuICBzb3J0YWJsZT8uZGVzdHJveSgpXG4gIHNvcnRhYmxlID0gbnVsbFxufSlcbjwvc2NyaXB0PlxuXG48c3R5bGU+XG4uZGVtby10YWJzID4gLmVsLXRhYnNfX2NvbnRlbnQge1xuICBwYWRkaW5nOiAzMnB4O1xuICBjb2xvcjogIzZiNzc4YztcbiAgZm9udC1zaXplOiAzMnB4O1xuICBmb250LXdlaWdodDogNjAwO1xufVxuPC9zdHlsZT4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgZ2V0Q3VycmVudEluc3RhbmNlIH0gZnJvbSAndnVlJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL3ByZXZpZXctMjA2NzAtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9kaXN0L2luZGV4LmNzcycsICdodHRwczovL3ByZXZpZXctMjA2NzAtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS90aGVtZS1jaGFsay9kYXJrL2Nzcy12YXJzLmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMjA2NzAtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9kaXN0L2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcInVuc3VwcG9ydGVkXCIsXG4gICAgXCJAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZVwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQGVsZW1lbnQtcGx1cy9pY29ucy12dWVAMi9kaXN0L2luZGV4Lm1pbi5qc1wiLFxuICAgIFwic29ydGFibGVqc1wiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vc29ydGFibGVqcy8rZXNtXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0yMDY3MC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguY3NzIn19)
